addon/utils/fetchBesluitTypes.js

```diff
@@ -1,75 +1,89 @@
-);
+export default async function fetchBesluitTypes(classificationUri, ENV) {
+  const query = `
+    PREFIX                    conceptscheme: <https://data.vlaanderen.be/id/conceptscheme/>
+    PREFIX                      BesluitType: <https://data.vlaanderen.be/id/concept/BesluitType/>
+    PREFIX              BesluitDocumentType: <https://data.vlaanderen.be/id/concept/BesluitDocumentType/>
+    PREFIX                             skos: <http://www.w3.org/2004/02/skos/core#>
+    PREFIX                              xsd: <http://www.w3.org/2001/XMLSchema#>
+    PREFIX                              rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+    PREFIX                             core: <http://mu.semte.ch/vocabularies/core/>
+    PREFIX                          besluit: <http://lblod.data.gift/vocabularies/besluit/>
+    PREFIX BestuurseenheidClassificatieCode: <http://data.vlaanderen.be/id/concept/BestuurseenheidClassificatieCode/>
+    PREFIX                              sch: <http://schema.org/>
+    PREFIX                             rule: <http://lblod.data.gift/vocabularies/notification/>
+    PREFIX                        lblodRule: <http://data.lblod.info/id/notification-rule/>
 
-async function readUrlIntoGraph(url) {
-  const besluitTypesGraph = new rdflib.NamedNode(
-    'http://data.lblod.info/besluitTypes'
-  );
-  const response = await fetch(url);
-  const text = await response.text();
-  const graph = rdflib.graph();
-  await rdflib.parse(text, graph, besluitTypesGraph.value, 'text/turtle');
-  return graph;
-}
-
-function findConceptsDecidableBy(graph, classificationUri) {
-  const classification = new rdflib.NamedNode(classificationUri);
-  const dataset = graph.match(null, SKOS_TOP_CONCEPT_OF, TYPE_CONCEPT_SCHEME);
-  return dataset
-    .filter((triple) =>
-      graph.holds(triple.subject, DECIDABLE_BY, classification)
-    )
-    .map((triple) => triple.subject);
-}
-
-function findPreflabel(graph, concept) {
-  return graph.anyValue(concept, SKOS_PREFLABEL, null);
+    SELECT ?s ?p ?o WHERE {
+      ?s skos:inScheme conceptscheme:BesluitType ;
+         besluit:notificationRule ?rule .
+      ?rule besluit:decidableBy <${classificationUri}> .
+      OPTIONAL { ?rule sch:validFrom ?validFrom . }
+      OPTIONAL { ?rule sch:validThrough ?validThrough . }
+      BIND(now() AS ?currentTime) .
+      BIND(STRLEN(STR(?validFrom)) > 0 AS ?validFromExists) .
+      BIND(STRLEN(STR(?validThrough)) > 0 AS ?validThroughExists) .
+      FILTER(
+        ((?validFromExists && ?validThroughExists) && (?currentTime < ?validThrough && ?currentTime >= ?validFrom)) ||
+        ((!?validFromExists && ?validThroughExists) && (?currentTime < ?validThrough)) ||
+        ((?validFromExists && !?validThroughExists) && (?currentTime >= ?validFrom))
+      ) .
+      OPTIONAL { ?s skos:prefLabel ?label . }
+      OPTIONAL { ?s skos:definition ?definition . }
+      OPTIONAL { ?s skos:broader ?parent . }
```

Comment:
Is this part of the query strictly necessary? As nothing is done with label, definition or parent.